drivers/hmis/app/models/hmis/hud/processors/client_processor.rb

```diff
@@ -132,8 +132,7 @@ def self.input_to_multi_fields(input_field, enum_map, not_collected_key, none_fi
       end
 
       primary = values.find { |v| v['primary'] == true }
-      # Build attributes for Client based on the Primary name. This already happens in the after_save hook on CustomClientName,
-      # but we need it here so that the correct values are present on the unpersisted record for validation.
+      # Build attributes for Client based on the Primary name
       client_attributes = if primary.present?
```

Comment:
I can't seem to comment on it, but 137 - 142 are roughly equivalent to `client.rb#426-430`, I wonder if it's worth DRYing that up a bit.

Reply:
Roughly, yes. They're different though, in this case the "primary name" that we're working with is not actually a CustomClientName object, it's just a hash map of values from the form submission.

Reply:
yeah, I noticed that, but the primary name object should respond just fine to `['first']`, etc.  If you want to clean that up....  Maybe just a comment to note that it would need to be changed there if this is ever changed.

Reply:
Ok, I added comments in both places